nixos/modules/services/logging/journaldriver.nix

```diff
@@ -0,0 +1,117 @@
+# This module implements a systemd service for running journaldriver,
+# a log forwarding agent that sends logs from journald to Stackdriver
+# Logging.
+#
+# It can be enabled without extra configuration when running on GCP.
+# On machines hosted elsewhere, the other configuration options need
+# to be set.
+#
+# For further information please consult the documentation in the
+# upstream repository at: https://github.com/aprilabank/journaldriver/
+
+{ config, lib, pkgs, ...}:
+
+with lib; let cfg = config.services.journaldriver;
+envVars = [
+  { name = "RUST_LOG"; value = cfg.logLevel; }
+  { name = "LOG_NAME"; value = cfg.logName; }
+  { name = "LOG_STREAM"; value = cfg.logStream; }
+  { name = "GOOGLE_CLOUD_PROJECT"; value = cfg.googleCloudProject; }
+  { name = "GOOGLE_APPLICATION_CREDENTIALS"; value = cfg.applicationCredentials; }
+];
+
+# Assemble an environment in which all values that have not been
+# specified explicitly are left unset:
+environment = with builtins; listToAttrs (filter (v: v.value != null) envVars);
+in {
+  options.services.journaldriver = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to enable journaldriver to forward journald logs to
+        Stackdriver Logging.
+      '';
+    };
+
+    logLevel = mkOption {
+      type = types.string;
+      default = "info";
```

Comment:
What valid values are there? You can change the type to `types.enum [ "info" "debug" ... ]` to only allow certain values.

Reply:
The values aren't constrained by an enum in this case, Rust's logging library accepts pretty complex configuration via this environment variable (for example one could enable trace logging for the HTTP client used within while not touching the log-level of the rest of the service).